File: arrays/majorityElemntInArray.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int candidate = 0;
        int count = 0;
        for (int currnum : nums) {
            if (count == 0) candidate = currnum;
            count += (currnum == candidate) ? 1 : -1;
        }
        return candidate;
    }
};

File: arrays/majorityElemntInArray.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        candidate = None
        count = 0
        for x in nums:
            if count == 0:
                candidate = x
            count += (1 if x == candidate else -1)
        return candidate
